modules/gateway/rpc_test.go

```diff
@@ -289,4 +271,15 @@ func TestBroadcast(t *testing.T) {
 	case <-time.After(10 * time.Millisecond):
 		// Neither peer should receive a broadcast.
 	}
+
+	// Test that calling broadcast with nil peers does not broadcast to g2 or g3.
+	g1.Broadcast("Recv", "foo", nil)
+	select {
+	case <-g2DoneChan:
+		t.Error("broadcast broadcasted to peers not in the peers arg")
+	case <-g3DoneChan:
+		t.Error("broadcast broadcasted to peers not in the peers arg")
+	case <-time.After(10 * time.Millisecond):
+		// Neither peer should receive a broadcast.
```

Comment:
this test covers the NOTE you wrote above, which further enforces the idea that the note isn't really needed.